docs/elasticsearch/search-template.md

```diff
@@ -0,0 +1,416 @@
+          "speaker": "KING HENRY IV",
+          "text_entry": "Find we a time for frighted peace to pant,"
+        }
+      }
+    ]
+  }
+}
+```
+
+If you have a stored template and want to validate it, you can use the `render` operation:
+
+```json
+POST _render/template
+{
+  "id": "play_search_template",
+  "params": {
+    "play_name": "Henry IV"
+  }
+}
+```
+
+#### Sample output
+
+```json
+{
+  "template_output": {
+    "from": "0",
+    "size": "10",
+    "query": {
+      "match": {
+        "play_name": "Henry IV"
+      }
+    }
+  }
+}
+```
+
+## Advanced parameter conversion with search templates
+
+You have a lot of different syntax options in Mustache to transpose the input parameters into a query.
+You can specify conditions, run loops, join arrays, convert arrays to JSON, and so on.
+
+### Conditions
+
+You can use the section tag in Mustache to represent conditions:
+
+```json
+{% raw %}{{#var}}var{{/var}}{% endraw %}
+```
+
+When `var` is a Boolean value, this syntax acts as an `if` condition. The `{% raw %}{{#var}}{% endraw %}` and `{% raw %}{{/var}}{% endraw %}` tags insert the values placed between them only if `var` evaluates to `true`.
+
+Using section tags would make your JSON invalid, so you must write your query in a string format instead.
+
+In this command, if you change the `limit` parameter to `true`, you would get back only two documents.
+That's because only when the condition is `true` the `size` parameter is activated.
+
+```json
+GET _search/template
+{
+  "source": "{% raw %}{ {{#limit}} \"size\": \"{{size}}\", {{/limit}}  \"query\":{\"match\":{\"play_name\": \"{{play_name}}\"}}}{% endraw %}",
+  "params": {
+    "play_name": "Henry IV",
+    "limit": false,
+    "size": 2
+  }
+}
+```
+
+You can also design an `if-else` condition.
+This command sets `size` to `2` if `limit` is `true` or sets it to `10` otherwise:
+
+```json
+GET _search/template
+{
+  "source": "{% raw %}{ {{#limit}} \"size\": \"2\", {{/limit}} {{^limit}} \"size\": \"10\", {{/limit}} \"query\":{\"match\":{\"play_name\": \"{{play_name}}\"}}}{% endraw %}",
+  "params": {
+    "play_name": "Henry IV",
+    "limit": true
+  }
+}
+```
+
+### Loops
+
+You can also use the section tag to implement a `foreach` loop:
+
+```
+{% raw %}{{#var}}{{.}}}{{/var}}{% endraw %}
+```
+
+When `var` is an array, the search template iterates through it and creates a `terms` query.
+
+```json
+GET _search/template
+{
+  "source": "{% raw %}{\"query\":{\"terms\":{\"play_name\":[\"{{#play_name}}\",\"{{.}}\",\"{{/play_name}}\"]}}}{% endraw %}",
+  "params": {
+    "play_name": [
+      "Henry IV",
+      "Othello"
+    ]
+  }
+}
+```
+
+This template is rendered as:
+
+```json
+GET _search/template
+{
+  "source": {
+    "query": {
+      "terms": {
+        "play_name": [
+          "Henry IV",
+          "Othello"
+        ]
+      }
+    }
+  }
+}
+```
+
+### Join
+
+You can use the `join` tag to concatenate values of an array separated by commas:
```

Comment:
Wording is confusing here. I think if you put (separated by commas) in parentheses, it's a little more clear, but use your discretion.